src/board_builder/board_builder.py

```diff
@@ -0,0 +1,258 @@
+        self._relative_pose_matrix = new_matrix
+
+    def _filter_data(self, data):
+        """ Filters the detector data to only keep the markers snapshots that appear in more than one detector """
+        label_counts = defaultdict(int)
+        for snapshots in data.values():
+            for snapshot in snapshots:
+                label_counts[snapshot.label] += 1
+
+        filtered_data = {}
+        for key, snapshots in data.items():
+            filtered_snapshots = [
+                snapshot for snapshot in snapshots
+                if label_counts[snapshot.label] >= 2
+            ]
+            if filtered_snapshots:
+                filtered_data[key] = filtered_snapshots
+        return filtered_data
+
+    def _find_matrix_input_index(self, pose_uuid, other_pose_uuid):
+        """ Given two pose uuids, return their index location in the relative pose matrix """
+
+        pose_index = -1
+        other_pose_index = -1
+
+        for index in self._index_to_marker_uuid:
+            if self._index_to_marker_uuid[index] == pose_uuid:
+                pose_index = index
+            if self._index_to_marker_uuid[index] == other_pose_uuid:
+                other_pose_index = index
+
+        if pose_index == -1:
+            raise ValueError(f"Pose UUID {pose_uuid} not found in the index mapping.")
+        if other_pose_index == -1:
+            raise ValueError(f"Pose UUID {other_pose_uuid} not found in the index mapping.")
+
+        return pose_index, other_pose_index
+
+    def _solve_pose(self, detector_data, timestamp):
+        """ Given marker ids and its corner locations, find its pose """
+        for detector_name in detector_data:
+            for markerSnapshot in detector_data[detector_name]:
+                if self.pose_solver.add_target_marker(int(markerSnapshot.label)):
+                    self._expand_matrix()
+
+        for detector_name in detector_data:
+            for markerSnapshot in detector_data[detector_name]:
+                corners_list: list[list[int]] = []
+                for corner in markerSnapshot.corner_image_points:
+                    corners_list.append([corner.x_px, corner.y_px])
+                marker_corners = MarkerCorners(
+                    detector_label=detector_name,
+                    marker_id=int(markerSnapshot.label),
+                    points=corners_list,
+                    timestamp=timestamp
+                )
+                self.pose_solver.add_marker_corners([marker_corners])
+
+        target_poses = self.pose_solver.get_target_poses()
+        self.target_poses = target_poses
+        visible_markers = []
+        for pose in target_poses:
+            visible_markers.append(pose.target_id)
+            if pose.target_id not in list(self._index_to_marker_uuid.values()):
+                self._index_to_marker_uuid[self._matrix_index] = pose.target_id
+                self._matrix_index += 1
```

Comment:
It would make sense to have the two blocks together. However, index_to_marker_uuid relies on having a uuid associated with each marker, and I can only get the uuid after running pose_solver. I could entirely change the mapping from index to marker label, but for now, it's too much reformatting just to refactor the code.

Reply:
See issue #74

Reply:
Yes, for where we're at, I think it's more valuable to merge all our changes and have a common base code to work from.